File: Algorithms/Searching/linearSearch.py
"""Linear Search"""

def linearSearch(array, n, x):
    for i in range(n):
        if (array[i] == x):
            return i
    return -1

if __name__ == "__main__":
    array = [1, 21, 1, 342, 12, 32, 213, ]
    x = 32
    n = len(array)
    # Apply the algorithm
    result = linearSearch(array, n, x)
    if (result == -1):
        print("Element is not present in the array\n")
    else:
        print("Element is present at index ", result, "\n")

File: Algorithms/Searching/linearSearch.cpp
/*Linear Search. O(n).*/
#include <iostream>
using namespace std;

int linearSearch(int arr[], int n, int x){
	int i;
	for (i = 0; i < n; i++){
		if (arr[i] == x){
			return i;
		}
	}
	return -1; // just if the the value is not present.
}

int main(void){
	int arr[] = { 2, 4, 5, 6, 7, 32, 53};
	int x = 32;
	int n = sizeof(arr) / sizeof(arr[0]);

	int index = linearSearch(arr, n, x);

	(index == -1)? cout << "The element is not present in the array."
		     : cout << "The element has been found at index " << index;
	
	return 0;
}


